src/validate.ts

```diff
@@ -29,7 +34,7 @@ function responseValidation(apiResponse: any, schema: any) {
  * @param responseType User defined intent response
  * @return Errors from AJV validation, if any. Undefined otherwise.
  */
-export function validate(apiResponse: any, responseType: 'sync' | 'query'|'execute'| 'disconnect') {
+export function validate(apiResponse: object, responseType: string) {
```

Comment:
I'm not sure that we should have `responseType` be downgraded to a `string` rather than keeping it within the pre-determined constants defined before. Using the set of constants allows the code to be more precise.

Reply:
Will be deprecated in the future for checking the intent type.